src/openfermion/ops/_qubit_operator.py

```diff
@@ -464,3 +464,20 @@ def __str__(self):
 
     def __repr__(self):
         return str(self)
+
+    def tracenorm(self):
+        """Calculate the trace norm (sqrt(Tr[O^O])) of the operator"""
+        if not self.terms:
+            return 0
+        norm2 = 0
+        for coeff in self.terms.values():
+            norm2 += numpy.abs(coeff)**2
+        return numpy.sqrt(norm2)
+
+    def renormalize(self):
+        """Fix the trace norm of an operator to 1"""
+        if not self.terms:
+            raise ZeroDivisionError
```

Comment:
I think it would be good to include a descriptive error message with this raise.  E.g. ZeroDivisionError('Cannot renormalize empty or 0 operator') and perhaps check for the same below at the same time after calculating norm, as I think a similar error could arise if the operator happened to have 0 coefficients.  I think we have compress routines that eliminate 0 operators, but they aren't always called automatically if I recall correctly.

Reply:
Sounds good. As there shouldn't be an error calculating the norm of a zero operator, I've shifted the error to after the norm is found, and added the message you suggested.